components/collection/drop/MintButton.vue

```diff
@@ -16,64 +16,50 @@
 
 <script setup lang="ts">
 import { NeoButton } from '@kodadot1/brick'
-import type { HolderOfCollectionProp } from '@/components/collection/drop/types'
 import useGenerativeDropMint, {
   useCollectionEntity,
 } from '@/composables/drop/useGenerativeDropMint'
 import { useDropStore } from '@/stores/drop'
 import { useDrop, useDropMinimumFunds } from '@/components/drops/useDrops'
-import holderOfCollectionById from '@/queries/subsquid/general/holderOfCollectionById.graphql'
 import { formatAmountWithRound } from '@/utils/format/balance'
-
-const props = defineProps<{
-  holderOfCollection?: HolderOfCollectionProp
-}>()
+import useHolderOfCollection from '@/composables/drop/useHolderOfCollection'
 
 const emit = defineEmits(['mint'])
 
 const { $i18n } = useNuxtApp()
-const { urlPrefix, client } = usePrefix()
-const { isLogIn, accountId } = useAuth()
+const { urlPrefix } = usePrefix()
+const { isLogIn } = useAuth()
 const { chainSymbol, decimals } = useChain()
 const dropStore = useDropStore()
 const { hasCurrentChainBalance } = useMultipleBalance()
 const { drop } = useDrop()
-const { mintCountAvailable, selectedImage, maxCount } = useGenerativeDropMint()
+const { mintCountAvailable, selectedImage, maxCount, mintedCount } =
+  useGenerativeDropMint()
 const { mintedAmountForCurrentUser } = useCollectionEntity()
 
 const { hasMinimumFunds } = useDropMinimumFunds()
-const { data: holderOfCollectionData } = await useAsyncData(
-  'holderOfCollectionData',
-  async () =>
-    await useAsyncQuery({
-      clientId: client.value,
-      query: holderOfCollectionById,
-      variables: {
-        id: drop.value?.holder_of,
-        account: accountId.value,
-      },
-    }).then((res) => res.data.value),
-  {
-    watch: [accountId, () => dropStore.runtimeMintCount],
-  },
-)
+const { availableNfts, holderOfCollection } = useHolderOfCollection()
 
 const isHolderAndEligible = computed(() => {
-  const maxMintLimit =
-    holderOfCollectionData.value?.nftEntitiesConnection?.totalCount || 0
-
-  const isHolder = maxMintLimit > 0
-
-  const hasNFTsAvailable = inject('hasNFTsAvailable', true) // true if not provided
+  const isHolder = mintedAmountForCurrentUser.value > 0
```

Comment:
Why change the check condition here? It's wrong because `isHolder` means the user has the required nft from `holderOfCollection`.

It's a little bit duplicated with this section. https://github.com/kodadot/nft-gallery/pull/9709/files#diff-ebe35e75569a918231b0ca2f8edb4d1ab28bdb6da55b7e5ec3fcf6881890832dR34
Could you pls unify them? (`hasNFTsAvailable` and `isHolder`)